lib/spack/spack/directory_layout.py

```diff
@@ -360,7 +360,8 @@ def remove_install_directory(self, spec, deprecated=False):
 
         elif os.path.exists(path):
             try:
-                shutil.rmtree(path)
+                shutil.rmtree(path, ignore_errors=False,
+                              onerror=fs.error_remove_readonly)
```

Comment:
This creates implied `ignore_errors=True` on non-Windows systems so:

* Update `error_remove_readonly` so it throws an error if it is ever called on a non-Windows system (to avoid similar issues sneaking by in the future)
* You'll need to do the same `kwargs` manipulation here as was done in `remove_linked_tree`

Reply:
I keep missing this change, thank you, my apologies. These changes have been made.